include/mra/misc/convolutiondata.h

```diff
@@ -204,6 +228,61 @@ namespace mra {
       }
     };
 
+  template <typename T, Dimension NDIM>
+  class Operator {
+
+  private:
+    size_type K;
+    std::map<Key<NDIM>, OperatorData<T, NDIM>> opdata;     // map for storing operator data
+    std::mutex cachemutex;                                 // mutex for thread safety
+
+    T norm_ns(Level n, const ConvolutionData<T>* ns[]) const {
+      T norm = 1.0, sum = 0.0;
+
+      for (size_type d = 0; d < NDIM; ++d) {
+        TensorView<T, 2> ns_rview(ns[d]->R);
+        auto ns_sview = ns[d]->S.current_view();
+        for (size_type i = 0; i < K; ++i) {
+          for (size_type j = 0; j < K; ++j) {
+            ns_rview(i, j) = 0.0;
+          }
+        }
```

Comment:
You're taking a const pointer and start modifying it. This probably should not be legal, i.e., the TensorView API should prohibit it. It's also not clear why you zero out `ns_rview` in the first place.